src/genai/extensions/langchain/embeddings.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import annotations
+
+from typing import List, Optional
+
+from pydantic import BaseModel, ConfigDict
+
+from genai._types import ModelLike
+from genai.client import Client
+from genai.text.embedding.embedding_service import CreateExecutionOptions
+from genai.text.embedding.schema import TextEmbeddingParameters
+
+__all__ = ["LangChainEmbeddingsInterface"]
+
+try:
+    from langchain_core.embeddings import Embeddings
+    from langchain_core.runnables.config import run_in_executor
+except ImportError:
+    raise ImportError("Could not import langchain: Please install ibm-generative-ai[langchain] extension.")  # noqa: B904
+
+
+class LangChainEmbeddingsInterface(BaseModel, Embeddings):
+    """
+    Class representing the LangChainChatInterface for interacting with the LangChain chat API.
+
+    Example::
+
+        from genai import Client, Credentials
+        from genai.extensions.langchain import LangChainEmbeddingsInterface
+        from genai.text.embedding import TextEmbeddingParameters
+
+        client = Client(credentials=Credentials.from_env())
+        embeddings = LangChainEmbeddingsInterface(
+            client=client,
+            model_id="sentence-transformers/all-minilm-l6-v2",
+            parameters=TextEmbeddingParameters(truncate_input_tokens=True)
+        )
+
+        embeddings.embed_query("Hello world!")
+        embeddings.embed_documents(["First document", "Second document"])
+    """
+
+    model_config = ConfigDict(extra="forbid", protected_namespaces=(), arbitrary_types_allowed=True)
```

Comment:
`arbitrary_types_allowed=True` needs to be set due to Pydantic not being able to resolve reference to the `Client`.

Reply:
Ok, I think it should be possible to do using the [private model attributes](https://docs.pydantic.dev/latest/concepts/models/#private-model-attributes), but this is also ok